pandas/core/dtypes/generic.py

```diff
@@ -57,6 +57,8 @@ def _check(cls, inst):
 ABCDateOffset = create_pandas_abc_type("ABCDateOffset", "_typ",
                                        ("dateoffset",))
 ABCInterval = create_pandas_abc_type("ABCInterval", "_typ", ("interval", ))
+ABCExtensionArray = create_pandas_abc_type("ABCExtensionArray", "_typ",
+                                           ("extension", "categorical",))
```

Comment:
isn't Categorical a first class EA?

Reply:
Yes, but we for Categorical we actually override the `_type` (what we say to external extension authors not to do), because we also have a `ABCCategorical` that needs that.